### PR TITLE
Simple maybe in the standard library

### PR DESCRIPTION
**Pull request by FredTheDino**
- Simple maybe in the standard library
- Fix minor bug with nested enums
- Add some maybe tests for the std.maybe

### CODE REVIEW

File: std/maybe.sy [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+Maybe :: enum
+    Just *
+    None
+end
+
+fromMaybe :: fn maybe, a ->
```
Comment (iwikal) [inline std/maybe.sy:7]: `maybe.orDefault` feels like a more descriptive name for this function.

**Review by iwikal — APPROVED** [review]

File: std/maybe.sy (hunk 1 above)
Comment (iwikal) [reply std/maybe.sy:7]: I see now that this is stolen from Haskell, while my suggestion is from Rust. I think the Rust name is better though :angel:

Comment (Quaqqer) [reply std/maybe.sy:7]: I think it is a lot more intuitive with or as well.

Comment (FredTheDino) [reply std/maybe.sy]: The Haskell function has the added context of saying what type you give in. I don't personally agree but I am being easily swayed today.

**Review by sornas — APPROVED** [review]

**Review by Quaqqer — APPROVED** [review]

**Pull request merged**